scipy/optimize/_cobyla_py.py

```diff
@@ -188,9 +191,9 @@ def fmin_cobyla(func, x0, cons, args=(), consargs=None, rhobeg=1.0,
 
 @synchronized
 def _minimize_cobyla(fun, x0, args=(), constraints=(),
-                     rhobeg=1.0, tol=1e-4, maxiter=1000,
-                     disp=False, catol=2e-4, callback=None, bounds=None,
-                     **unknown_options):
+                     rhobeg=1.0, tol=1e-6, maxiter=1000,
```

Comment:
Does decreasing `tol` improve the performance on `COBYLA` on our benchmarks? Strictly speaking, this is a backwards compatability break.

Reply:
I'll take a look as how this impacts performance. I was trying to remember why I changed it, and I think the reason is that 1e-6 is the [default for PRIMA](https://github.com/nbelakovski/prima/blob/main/fortran/common/consts.F90#L217) (note that PRIMA refers to this variable as rhoend, i.e. the final radius of the trust region), so since PRIMA was built and evaluated with this as a default it made sense to keep it consistent, but I'll run those benchmarks and we'll see what comes up.

Reply:
If this is the default for PRIMA, that is already a strong enough argument for me to change it in future. We should rely on the authors of PRIMA wherever possible.

We have to wait two minor versions though per our usual deprecation process.

Reply:
I started doing more evaluations, and I discovered that the benchmarks override the tolerance to 1e-4, so the previous result I posted was actually comparing old COBLYA vs new COBYLA against each other at a tolerance of 1e-4. I modified the new code to ignore the provided argument and just hardcode the tolerance and ran the benchmarks again. I debated whether or not it made sense to compare PRIMA COBYLA at various tolerances against the old COBYLA fixed at 1e-4, and ultimately I decided that it did make sense because our goal right now is not to compare these two implementations in a scientific sense, but to replace one with the other and potentially replace default arguments as well.

The data is attached, and below I've summarized how many cases are worse/better for min_obj/mean_nfev. Unsurprisingly, the mean_nfev gets higher as the tolerance is made tighter.

|   1e-4 vs 1e-4    |  min_obj | mean_nfev |
|-|-|-|
| worse  |    11    |     11    |
| better |    23    |     16    |

|   1e-4 vs 1e-5    |  min_obj | mean_nfev |
|-|-|-|
| worse  |    7     |     19    |
| better |    25    |     2     |

|  1e-4 vs 1e-6     |  min_obj | mean_nfev |
|-|-|-|
| worse  |    7     |     20    |
| better |    26    |     1     |

|   1e-4 vs 1e-8     |  min_obj | mean_nfev |
|-|-|-|
| worse  |    7     |     22    |
| better |    26    |     0     |

Given this data I'd say changing the tolerance from the current (as implemented in this PR) 1e-6 back down to 1e-4. Even at 1e-5 mean_nfev starts to get significantly higher, but at 1e-4 it seems like there's a nice tradeoff where both min_obj and mean_nfev are better (by number of cases). This is a change from what I wrote in my previous comment but I think the data supports it. Thoughts?

[1e-4.txt](https://github.com/user-attachments/files/16830877/1e-4.txt)
[1e-5.txt](https://github.com/user-attachments/files/16830878/1e-5.txt)
[1e-6.txt](https://github.com/user-attachments/files/16830879/1e-6.txt)
[1e-8.txt](https://github.com/user-attachments/files/16830880/1e-8.txt)

Reply:
Thanks for the deep dive. I agree, let's set it to 1e-4.